models.py

```diff
@@ -316,12 +346,28 @@ def _compute_emissions(self,
             # shape: (batch_size, seq_length, total_suffix_emb_size)
             affix_features = embedded_suffixes
 
-        # shape: (batch_size, seq_length, (2*window + 1) * word_embedding_size)
-        inputs = word_features
         if affix_features is not None:
             # shape: (batch_size, seq_length, (2*window+1)*word_embedding_size+total_affix_emb_size)
             inputs = torch.cat((inputs, affix_features), dim=-1)
 
+        # apply Bidirectional LSTM
+        if self.uses_lstm:
+            if mask is None:
+                # shape (batch_size, seq_length)
+                mask = self._get_mask_for(words)   
+            
+            seq_lengths = torch.sum(mask.int(), dim=1)
+            seq_lengths, sent_perm = seq_lengths.sort(0, descending=True)
+            inputs = inputs[sent_perm]
+
+            packed_input = pack_padded_sequence(inputs, seq_lengths.data.tolist(), batch_first=True)
```

Comment:
Can just `.data` works? Calling `.tolist()` is *very* slow, up to 1000x times slower than `.numpy()`. If just `.data` works, use that and no need to call `.numpy()`. If it doesn't, then `.numpy()` is preferred as it's faster.